ecl2df/summary.py

```diff
@@ -39,34 +42,56 @@ def date_range(start_date, end_date, freq):
     Returns:
         list of datetimes
     """
-    if freq in PD_FREQ_MNEMONICS:
-        freq = PD_FREQ_MNEMONICS[freq]
-    return pd.date_range(start_date, end_date, freq=freq)
+    return pd.date_range(start_date, end_date, freq=PD_FREQ_MNEMONICS.get(freq, freq))
+
+
+def _ensure_date_or_none(some_date):
+    """Ensures an object is either a date or None
+
+    Args:
+        some_date: string or a datetime.date
 
+    Returns:
+        datetime.date: None if input is None.
 
-def normalize_dates(start_date, end_date, freq):
+    Raises:
+        TypeError: if input is not None and not a date
     """
-    Normalize start and end date according to frequency
-    by extending the time range.
+    if some_date:
+        if isinstance(some_date, str):
+            return dateutil.parser.parse(some_date).date()
+        if not isinstance(some_date, dt.date):
+            raise TypeError(f"Not a date type: {str(some_date)}")
+    return some_date
 
-    So for [1997-11-05, 2020-03-02] and monthly frequency
-    this will transform your dates to
-    [1997-11-01, 2020-04-01]
 
-    For yearly frequency it will return [1997-01-01, 2021-01-01].
+def _crop_datelist(eclsumsdates, freq, start_date, end_date):
+    """Helper function for resample_smry_dates, taking care of
+    the special cases where the list of dates should not be resampled, but
+    only cropped or returned as is.
 
-    Args:
-        start_date: datetime.date
-        end_date: datetime.date
-        freq: string with either 'monthly' or 'yearly'.
-            Anything else will return the input as is
-    Return:
-        Tuple of normalized (start_date, end_date)
+    Arguments are the same as for resample_smry_dates
     """
-    if freq in PD_FREQ_MNEMONICS:
-        freq = PD_FREQ_MNEMONICS[freq]
-    offset = pd.tseries.frequencies.to_offset(freq)
-    return (offset.rollback(start_date).date(), offset.rollforward(end_date).date())
+    if freq == "raw":
```

Comment:
If parameters to `_drop_datelist` are the same as `resample_smry_dates` then these magic strings should really be shared between the two.

What exactly happens when `freq` is a date/datetime?

Reply:
magic strings replaced by module constants. 

if `freq` is a datetime, it is caught in the `if`-statement ~6 lines below.